_posts/en/newsletters/2022-04-13-newsletter.md

```diff
@@ -0,0 +1,171 @@
+    original MuSig proposal (now called *MuSig1*).  MuSig2 is easier to
+    implement than the alternative MuSig-DN (deterministic nonce),
+    although there are tradeoffs between MuSig2 and MuSig-DN that some
+    application developers may wish to consider when choosing which
+    protocol to use.
+
+## Bitcoin Core PR Review Club
+
+*In this monthly section, we summarize a recent [Bitcoin Core PR Review Club][]
+meeting, highlighting some of the important questions and answers.  Click on a
+question below to see a summary of the answer from the meeting.*
+
+[Prevent block index fingerprinting by sending additional getheaders messages][]
+is a PR by Niklas Gögge to prevent a node from being fingerprinted based on its
+block index.
+
+{% include functions/details-list.md
+
+  q0="What is the block index and what is it used for?"
+  a0="The block index is an in-memory index for looking up block headers and the
+locations of block data on disk. It may keep multiple branches (i.e. including
+stale block headers) of the block 'tree' to accommodate reorgs."
+  a0link="https://bitcoincore.reviews/24571#l-17"
+
+  q1="Should we keep stale blocks in the block index? Why or why not?"
+  a1="When multiple branches exist, keeping them indexed allows the node to
+quickly switch branches if the most-work chain changes.  Some participants noted
+that it may not be very useful to keep very old stale blocks, since the
+likelihood of a reorg is extremely low. However, these headers use very little
+storage space and, since nodes check the Proof of Work before storing them,
+sending valid-PoW stale headers in the hopes of exhausting nodes' resources
+would be disproportionately expensive."
+  a1link="https://bitcoincore.reviews/24571#l-68"
+
+  q2="Describe the attack using a node's block index for fingerprinting."
+  a2="During IBD, a node only requests and downloads blocks belonging to the
+most-work chain it learned about during the initial headers sync. As such, the
+stale blocks in its block index were usually mined after IBD, but this may vary
+naturally or be manipulated by an attacker with a large collection of past stale
+headers. An attacker with a stale branch of headers H and H+1 can send H+1 to a
+node. If the node doesn't have H+1's predecessor, H, in its block index, it will
+request H using a `getdata` message. If it already has H, it won't request it."
+  a2link="https://bitcoincore.reviews/24571#l-75"
+
+  q3="Why is it important to prevent node fingerprinting?"
+  a3="Node operators may employ various techniques to obfuscate their node's IP
+address, e.g. by using Tor.  However, the privacy benefits may be limited or
+negated if attackers can link the IPv4 and Tor addresses of a node running on
+both networks. If a node is running on Tor only, fingerprinting could be used to
+link multiple Tor addresses belonging to the same node, or identify the node
+if/when it switches to IPv4."
+  a3link="https://bitcoincore.reviews/24571#l-84"
```

Comment:
@glozow: It seems like it may not have gotten covered in the meeting, but I'm kinda left hanging with the question how the attack is prevented by the PR. Looking at the PR itself, it seems that the defense is that the node will always request the stale header in order to not reveal whether or not it has seen it before in order to prevent fingerprinting. Perhaps this should be mentioned in the context of the Review Club summary.

Reply:
I agree that the summary does leave the reader hanging about the solution, and also that the solution doesn't seem to be mentioned in the actual meeting, so I'm going to leave this as an option for @glozow to add a description if she wants or we can leave the section as is---it's ok IMO if people need to DYOR to find the solution like @Xekyo did.